src/main/java/gmailgdogra/WriteOutputToXlsx.java

```diff
@@ -53,99 +45,111 @@
-        System.out.println("loc = " + loc);
-        for (SwipeRecord record : records) {
-            String location = loc.toString();
-            String firstName = record.getOfficer().getFirstName();
-            String lastName = record.getOfficer().getLastName();
-            String eventDate;
-            String logicalDevice;
-            LocalDateTime eventDateTime = record.getSwipeDateTime();
-            if (eventDateTime == null) {
-                eventDate = "Swipe Missing";
-                logicalDevice = "";
-            } else {
-                eventDate = eventDateTime.toString();
-                logicalDevice = record.getDeviceName();
+    private static void applyStyles(Sheet sheet) {
+        XSSFCellStyle titleRowStyle = createTitleRowStyle();
+        XSSFCellStyle highlightedRowStyle = createHighlightedRowStyle();
+        XSSFCellStyle normalRowStyle = createNormalRowStyle();
+
+        for (Row row : sheet) {
+            int currentRowNo = row.getRowNum();
+            if (currentRowNo == 0) {
+                applyStyle(currentRowNo, titleRowStyle);
+                continue;
             }
-            OutputRow data = OutputRow.of(location, firstName, lastName, eventDate, logicalDevice);
-            addRow(sheet, data);
-        }
-    }
 
-    private static List<SwipeRecord> getSwipesForGivenLoc(List<SwipeRecord> outputData, List<Shift> shifts,
-                                                          Location location) {
-        List<SwipeRecord> returnList = new ArrayList<>();
-        for (Shift shift : shifts) {
-            if (shift.getLocation() == location) {
-                Officer shiftOfficer = shift.getOfficer();
-                returnList.addAll(outputData.stream()
-                        .filter(swipe -> shiftOfficer.equals(swipe.getOfficer()))
-                        .collect(Collectors.toList()));
+            String swipeEventCellValue = row.getCell(EVENT_DATE_COL_NO).getStringCellValue();
+            if ("No Swipe Found".equals(swipeEventCellValue) ||
+                    isLateSwipeIn(currentRowNo)) {
+                applyStyle(currentRowNo, highlightedRowStyle);
+            } else {
+                applyStyle(currentRowNo, normalRowStyle);
             }
         }
-        return returnList;
     }
 
-    private static void autoSizeColumns(Sheet sheet) {
-        for (int column = 0; column < 5; column++) {
-            sheet.autoSizeColumn(column);
-        }
+    private static boolean isLateSwipeIn(int rowNo) {
+
+        Row row = workbook.getSheetAt(SHEET_INDEX).getRow(rowNo);
+        String locationCellVal = row.getCell(LOCATION_COL_NO).getStringCellValue();
+        String timeInfo = row.getCell(EVENT_DATE_COL_NO).getStringCellValue().substring(TIME_INFO_START_INDEX);
+        String logicalDevCellVal = row.getCell(LOGICAL_DEVICE_COL_NO).getStringCellValue();
+        LocalTime swipeInTime = LocalTime.parse(timeInfo);
+
+        if (logicalDevCellVal.endsWith("OUT")) {
```

Comment:
which jdk version do you use? since java12 there is a new switch statement which you should use in such cases

Reply:
currently using 11